### ISSUE TITLE
HTTPException duplicates headers instead of overwriting

### ISSUE THREAD

**Issue by mark0978** [Pylons/pyramid#2749]
If you pass a content-type as a header to HTTPException, you'd expect the content type to change, instead you get 2 content-type header lines, the one that is injected for you, then the one you sent to the __init__

```
HTTP/1.1 404 Bad Request
Content-Length: 32
Content-Type: text/html; charset=UTF-8
Content-Type: application/json; charset=UTF-8
Date: Wed, 17 Aug 2016 14:06:16 GMT
```

This leaves no easy way to change the content-type of a response.  It is because of this line:  https://github.com/Pylons/pyramid/blob/master/pyramid/httpexceptions.py#L225
which simply extends the headers rather than replacing them as update would have done

Also, on line 203,  the comment says ```request body``` when it should say ```response body```

**Comment by mmerickel** [comment]
You're not wrong, but I think you can easily get around this one by passing the `content_type='application/json'` arg to the class instead of as a raw header.

**Comment by mark0978** [comment]
You are correct, maybe this is just a problem with documentation?  It's pretty subtle to know you can pass content_type= to an exception which will end up going to the Response.__init__()

I was reading the docstrings in the method itself.

**Comment by mmerickel** [comment]
Yeah they are `Response` subclasses. Maybe you could submit a PR that makes this a little bit more clear to you. I'm surprised myself that this happens and I can't think of a great reason why it would extend instead of update but I'm pretty hesitant to change it for compatibility reasons.

**Comment by mmerickel** [comment]
Actually I just realized I think the reason it does an extend is because a common use-case is setting cookies, and thus multiple `Set-Cookie` headers are added. An update would not allow that to work.

**Comment by mark0978** [comment]
Done

**Comment by mmerickel** [comment]
Closed via #2750.
